### PR TITLE
Refactor: Refactor workspace

### PR DESCRIPTION
**Pull request by wen-qing-255**

### CODE REVIEW

File: src/app/flux/appbar-menu/index.js [hunk 1]
```diff
@@ -295,6 +295,9 @@ export const actions = {
                         break;
                 }
             });
+            editMenu.submenu.forEach(item => {
```
Comment (xiekun1992) [inline src/app/flux/appbar-menu/index.js:298]: Is `Copy & Paste` functions available after the code

File: src/server/controllers/Marlin/Marlin.js [hunk 2]
```diff
@@ -293,13 +296,29 @@ class MarlinReplyParserPurifierNotEnabled {
...
+        //     type: MarlinReplyParserPurifierNotEnabled
+        // };
+        if (r2 || r3) {
+            return {
+                type: MarlinReplyParserPurifierNotEnabled,
+                payload: {
+                    airPurifierHasPower: false
+                }
+            };
+        } else if (r0 || r1) {
```
Comment (bbaa77770) [inline src/server/controllers/Marlin/Marlin.js:312]: Add a new type?

Comment (bbaa77770) [reply src/server/controllers/Marlin/Marlin.js:312]: And payload is not used

File: src/app/ui/widgets/Connection/WifiConnection.jsx [hunk 3]
```diff
@@ -309,19 +426,28 @@ function WifiConnection() {
...
                         style={{
-                            marginTop: '10px'
+                            marginTop: '8px'
                         }}
-                    />
-                    {headType === HEAD_PRINTING && <PrintingState headType={headType} />}
-                    {headType === HEAD_LASER && <LaserState headType={headType} />}
+                    /> */}
+                    {/* {headType === HEAD_PRINTING && <PrintingState headType={headType} />}
+                    {headType === HEAD_LASER && <LaserState headType={headType} />} */}
```
Comment (jane-rose) [inline src/app/ui/widgets/Connection/WifiConnection.jsx]: delete

File: src/app/ui/widgets/Marlin/JogDistance.jsx [hunk 4]
```diff
@@ -84,7 +84,7 @@ class JogDistance extends PureComponent {
         return (
             <div className={styles['jog-distance-control']}>
                 <div className="input-group input-group-sm">
-                    <div className="input-group-btn">
+                    <div className="input-group-btn" style={{ display: 'flex' }}>
```
Comment (jane-rose) [inline src/app/ui/widgets/Marlin/JogDistance.jsx:87]: using classname 'sm-flex"

**Review by jane-rose — APPROVED** [review]

File: src/app/resources/i18n/de/resource.json [hunk 5]
```diff
@@ -1064,4 +1090,4 @@
     "key-unused-Z Offset": "Z-Ausgleich",
     "key-unused-{{pageLength}} per page": "{{pageLength}} pro Seite",
     "key_ui/widgets/CNCPath/CNCPath_Transformation": "Umwandlung"
-}
+}
```
Comment (xiekun1992) [inline src/app/resources/i18n/de/resource.json:1093]: add newline at the end of file

**Review by xiekun1992 — APPROVED** [review]

**Pull request merged**